### ISSUE TITLE
Support customer-managed keys for server_side_encryption in DynamoDB

### ISSUE THREAD

**Issue by sworisbreathing** [terraform-providers/terraform-provider-aws#8137]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As of November, 2018, [CreateTable](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_CreateTable.html) in DynamoDB now supports [Server-Side Encryption](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_SSESpecification.html) with the use of a specific customer-managed key (as opposed to the default DynamoDB KMS master key). However, at the moment, the `aws_dynamodb_table` resource [only supports turning SSE on/off](https://github.com/terraform-providers/terraform-provider-aws/blob/v2.4.0/aws/resource_aws_dynamodb_table.go#L251) but does not support specifying the KMS master key ID used for SSE.

Ideally, we should be able to pick which KMS key is used for encrypting data in a DDB table through the `aws_dynamodb_table` resource.

In my example below, I've also included `sse_algorithm` as I've copied it from how `aws_s3_bucket` works. It should be noted that per the documentation for `SSESpecification`, only KMS is applicable. So it's debatable whether or not terraform should expose `sse_algorithm` as a config option, given that only kms appears to be supported.

### New or Affected Resource(s)

* aws_dynamodb_table

### Potential Terraform Configuration

```hcl
resource "aws_kms_key" "mykey" {}

resource "aws_dynamodb_table" "mytable" {
  ...
  server_side_encryption {
    enabled = true
    kms_master_key_id = "${aws_kms_key.mykey.arn}"
    sse_algorithm     = "aws:kms"
  }
}
```

### References

* [CreateTable API Documentation](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_CreateTable.html)

* [SSESpecification for CreateTable API](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_SSESpecification.html)

* [How Amazon DynamoDB Uses AWS KMS](https://docs.aws.amazon.com/kms/latest/developerguide/services-dynamodb.html)

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/3298
* https://github.com/terraform-providers/terraform-provider-aws/issues/7498

**Comment by ewbankkit** [comment]
@sworisbreathing Thanks for this; We saw this change in the DynamoDB API about 8 months ago in the release notes for [AWS SDK v1.15.16](https://github.com/aws/aws-sdk-go/releases/tag/v1.15.16):

> service/dynamodb: Updates service API and documentation
> Added SSESpecification block to update-table command which allows users to modify table Server-> Side Encryption. Added two new fields (SSEType and KMSMasterKeyId) to SSESpecification block used by create-table and update-table commands. Added new SSEDescription Status value UPDATING

and I did some work in https://github.com/terraform-providers/terraform-provider-aws/pull/5666 that I ultimately ended up closing as at the time the DynamoDB service didn't support the additions.
I still have the code so I will see if the service has been updated.

**Comment by ewbankkit** [comment]
This has now been officially [announced](https://aws.amazon.com/about-aws/whats-new/2019/11/encrypt-your-amazon-dynamodb-data-by-using-your-own-encryption-keys/).
I will try and reuse my code from https://github.com/terraform-providers/terraform-provider-aws/pull/5666.

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 2.47.0 of the Terraform AWS Provider, tomorrow. Thanks to @ewbankkit for the implementation. 👍
